plugins/inputs/mesos/mesos_test.go

```diff
@@ -683,12 +683,66 @@ func TestCreateHTTPClientIAM(t *testing.T) {
 	require.NotNil(client)
 	require.NotNil(client.Transport)
 
-	// test that all requests on client send an Authorization header with token
+	// test that all requests on client send an Authorization header with token and default User-Agent header
 	tokenTestServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		auth := r.Header.Get("Authorization")
 		if auth != "token="+testToken {
 			t.Fatalf("Expected request header: `Authorization: token=%s`. Got: %s", testToken, auth)
 		}
+		userAgent := r.Header.Get("User-Agent")
+		if userAgent != "dcos-go" {
+			t.Fatalf("Expected request header: `User-Agent: dcos-go`. Got: %s", userAgent)
+		}
```

Comment:
This test is now verifying more than IAM, which makes its name `TestCreateHTTPClientIAM()` misleading.

Reply:
Reformatted all the `createHttpClient()` tests into tables structure